### PR TITLE
Gimbal v2: implement take_control and release_control

### PR DESCRIPTION
**Pull request by JonasVautherin**

### CODE REVIEW

File: src/plugins/gimbal/gimbal_protocol_v1.cpp [hunk 1]
```diff
@@ -156,4 +156,56 @@ void GimbalProtocolV1::set_roi_location_async(
...
+    std::thread([this, &callback]() {
+        std::unique_lock<std::mutex> lock(_control_thread_mutex);
+
+        while (_control_callback) {
+            _system_impl.call_user_callback(
+                [this, callback]() { callback(_current_control_status); });
+            _control_thread_cv.wait_for(lock, std::chrono::seconds(1));
+        }
+    });
+}
```
Comment (julianoes) [inline src/plugins/gimbal/gimbal_protocol_v1.cpp:209]: I don't really get that. I assume it's overkill that we could just ignore. I guess we'll leave it in now and clean it up later.

**Review by julianoes — APPROVED** [review]
Works for me against https://github.com/PX4/PX4-SITL_gazebo/pull/703 :tada:

File: src/plugins/gimbal/gimbal_protocol_v1.cpp (hunk 1 above)
Comment (JonasVautherin) [reply src/plugins/gimbal/gimbal_protocol_v1.cpp:209]: I just want to allow an application to use the same logic for both v1 and v2. So you need to get some status updates if you expect them, right? :blush:

Comment (julianoes) [reply src/plugins/gimbal/gimbal_protocol_v1.cpp:209]: Hm, fair enough. If it works you can leave it in :)

**Pull request merged**